docs/actions-new.md

```diff
@@ -0,0 +1,72 @@
+<!--
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+OpenWhisk supports [several languages and runtimes](actions.md#languages-and-runtimes) but
+there may be other languages or runtimes that are important for your organization, for which
+you want tighter integration with the platform. Adding a new language, or runtime with bespoke
+packages and third party dependencies is the same from the OpenWhisk platform which is language
+and runtime agnostic.
+
+The canonical unit of execution is a container which implements a specific interface:
```

Comment:
we'll need to document this interface...

Reply:
Would be good to express the required interface in a precise way.

Now digressing a bit ... Going forward I would like to prototype a way where runtime implementations are passed a URL of the attachment and they directly pull the attachment from storage instead of invoker acting as a proxy. For e.g. each runtime can just package curl and upon receiving the url just do a get via curl and stream it to file system. This would be more of optional capability supported by runtime.

So should we have some kind of versioning of the interface now where Invoker can detect what all a runtime t supports and based on that talks to it in a compatible way. For e.g. one way to implement the above change would be to add some capability attribute in runtime manifest and then based on that use a different approach (say use /init/v2) to talk to runtime.

Reply:
a reason we haven't documented the interface was to retain the ability to make breaking changes... but maybe that ship has sailed. so i've gone ahead and added it here... might be the first time we really make this official.